dafni/README.md

```diff
@@ -10,8 +10,8 @@ to upload the framework onto [DAFNI](https://www.dafni.ac.uk).
 - `data` contains two sub-folders (the structure is important for DAFNI).
   - `inputs` is a folder that contains the input files that are (separately) uploaded to DAFNI.
     - `causal_tests.json` is a JSON file that contains the causal tests.
-    - `variables.json` is a JSON file that contains the variables and constraints to be used.
-    - `dag.dot` is a dot file that contains the directed acyclc graph (dag) file.
+    - `dag.dot` is a dot file that contains the directed acyclc graph (dag) file, 
+       including the variables (and constraints if necessary) as node attributes.
```

Comment:
How are constraints expressed as node attributes? I thought constraints came in the causal tests JSON file?

Reply:
@jmafoster1 I didn't realise `constraints` are defined in the causal tests file. In that case, we'll need something separate to parse the JSON file for the constraints. 

Do we have an example JSON file that I can use that uses constraints in the causal tests file? I can only find one in the Poisson example that [defines constraints in the script](https://github.com/CITCOM-project/CausalTestingFramework/blob/bb2527b01c1382e960b96ad637fce6e3efe3770e/examples/poisson/example_run_causal_tests.py#L118) but doesn't actually use it in the [JSON file](https://github.com/CITCOM-project/CausalTestingFramework/blob/main/examples/poisson/causal_tests.json) for some reason, which doesn't make sense to me.

Reply:
Yes, see https://github.com/CITCOM-project/CausalTestingFramework/pull/299#discussion_r1880022545

The constraints in the script are an unused variable, so I guess they weren't transfered to the JSON file. We probably put them in at some point because they were mandatory and then made them optional but didn't bother to get rid of the variable. This stuff was written a LONG time ago before we had linting for PRs, so I guess it got missed.